frontend/stmt_inferrer.py

```diff
@@ -296,12 +279,30 @@ def _init_func_context(args, context, solver):
     return local_context, args_types
 
 
+def annotated(node):
+    """Check the arguments and return are annotated in a function definition"""
+    if not node.returns:
+        return False
+    for arg in node.args.args:
+        if not arg.annotation:
+            return False
+    return True
+
+
 def _infer_func_def(node, context, solver):
     """Infer the type for a function definition"""
+    if annotated(node):
+        return_annotation = node.returns
+        args_annotations = []
+        for arg in node.args.args:
+            args_annotations.append(arg.annotation)
+        context.annotated_functions[node.name] = (args_annotations, return_annotation)
+        return
```

Comment:
I don't think we should do this. This means that we don't type check annotated functions any more and we can only statically resolve them. I'd consider both of those disadvantages. I think we should only do this if the function is a stub and/or uses type parameters.